src/hotspot/share/services/diagnosticArgument.hpp

```diff
@@ -59,6 +60,11 @@ class MemorySizeArgument {
   char _multiplier;
 };
 
+class FileArgument {
+  public:
+    char *_name;
+};
```

Comment:
Note that the reason for the leak is probably the fact that you don't clear old values on parse_value. See e.g. how char* does it. However, since you allocate with a constant size anyway, the buffer size never changes, you could just as well either follow my advice above (inlining), or just re-use the existing pointer.

Reply:
Hi Thomas, 

Yes - this was an oversight on my end. I was not directly calling the `destroy_value()` function. I tried to follow more closely what’s done for `char*`, as I like the consistency throughout. 

I did a quick check and I don’t see any more leaks with NMT. Does the new change make sense to you as well? 

Thank you for the feedback!